test/kubemark/gce/util.sh

```diff
@@ -31,127 +31,54 @@ function authenticate-docker {
-    --size "${MASTER_DISK_SIZE}" &
-
-  if [ "${EVENT_PD:-}" == "true" ]; then
-    run-gcloud-compute-with-retries disks create "${MASTER_NAME}-event-pd" \
-      "${GCLOUD_COMMON_ARGS[@]}" \
-      --type "${MASTER_DISK_TYPE}" \
-      --size "${MASTER_DISK_SIZE}" &
-  fi
-
-  get-or-create-master-ip &
-
-  wait
-
-  run-gcloud-compute-with-retries instances create "${MASTER_NAME}" \
-    "${GCLOUD_COMMON_ARGS[@]}" \
-    --address "${MASTER_IP}" \
-    --machine-type "${MASTER_SIZE}" \
-    --image-project="${MASTER_IMAGE_PROJECT}" \
-    --image "${MASTER_IMAGE}" \
-    --tags "${MASTER_TAG}" \
-    --subnet "${SUBNETWORK:-${NETWORK}}" \
-    --scopes "storage-ro,logging-write" \
-    --boot-disk-size "${MASTER_ROOT_DISK_SIZE}" \
-    --disk "name=${MASTER_NAME}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no"
-
-  run-gcloud-compute-with-retries instances add-metadata "${MASTER_NAME}" \
-    "${GCLOUD_COMMON_ARGS[@]}" \
-    --metadata-from-file startup-script="${KUBE_ROOT}/test/kubemark/resources/start-kubemark-master.sh" &
-
-  if [ "${EVENT_PD:-}" == "true" ]; then
-    echo "Attaching ${MASTER_NAME}-event-pd to ${MASTER_NAME}"
-    run-gcloud-compute-with-retries instances attach-disk "${MASTER_NAME}" \
-   "${GCLOUD_COMMON_ARGS[@]}" \
-    --disk "${MASTER_NAME}-event-pd" \
-    --device-name="master-event-pd" &
-  fi
-
-  run-gcloud-compute-with-retries firewall-rules create "${MASTER_NAME}-https" \
-    --project "${PROJECT}" \
-    --network "${NETWORK}" \
-    --source-ranges "0.0.0.0/0" \
-    --target-tags "${MASTER_TAG}" \
-    --allow "tcp:443" &
-
-  run-gcloud-compute-with-retries firewall-rules create "${MASTER_NAME}-internal" \
-    --project "${PROJECT}" \
-    --network "${NETWORK}" \
-    --source-ranges "10.0.0.0/8" \
-    --target-tags "${MASTER_TAG}" \
-    --allow "tcp:1-2379,tcp:2382-65535,udp:1-65535,icmp" &
-
-  wait
-}
-
-# Command to be executed is '$1'.
-# No. of retries is '$2' (if provided) or 1 (default).
-function execute-cmd-on-master-with-retries() {
-  RETRIES="${2:-1}" run-gcloud-compute-with-retries ssh "${MASTER_NAME}" --zone="${ZONE}" --project="${PROJECT}" --command="$1"
+function create-kubemark-master {
+  # We intentionally override env vars in subshell to preserve original values.
+  # shellcheck disable=SC2030,SC2031
+  (
+    export KUBECONFIG="${RESOURCE_DIRECTORY}/kubeconfig.kubemark"
+    export CLUSTER_NAME="${CLUSTER_NAME}-kubemark"
+    export KUBE_CREATE_NODES=false
+    export KUBE_GCE_INSTANCE_PREFIX="${KUBE_GCE_INSTANCE_PREFIX}-kubemark"
+    
+    # Even if the "real cluster" is private, we shouldn't manage cloud nat.
+    export KUBE_GCE_PRIVATE_CLUSTER=false
+
+    # Quite tricky cidr setup: we set KUBE_GCE_ENABLE_IP_ALIASES=true to avoid creating
+    # cloud routes and RangeAllocator to assign cidrs by kube-controller-manager.
+    export KUBE_GCE_ENABLE_IP_ALIASES=true
+    export KUBE_GCE_NODE_IPAM_MODE=RangeAllocator
+    
+    # Disable all addons.
```

Comment:
Can you add a comment that we enable (most of?) them by running dedicated pod in the base cluster?

Reply:
Done.